### DISCUSSION TITLE
Add page-delimiter for elisp source files

### DISCUSSION THREAD

**Discussion by phil-s** [alphapapa/ement.el#312] [Ideas]
Some of the files are quite large, but they are already organised into logical sections with commented headings, which means it would be very easy to make those sections into pages by either adding a traditional page break `^L` character between sections, or by doing something like the following:

```diff
modified   .dir-locals.el
@@ -2,4 +2,5 @@
 ;;; For more information see (info "(emacs) Directory Variables")
 
 ((emacs-lisp-mode . ((fill-column . 90)
-		     (indent-tabs-mode . nil))))
+		     (indent-tabs-mode . nil)
+		     (page-delimiter . "^;;;;"))))
```

That then lets use navigate by page, and narrow to page, both of which are useful abilities.

**Comment by alphapapa** [comment]
If it's helpful to other developers, I don't object to adding the page breaks where appropriate.  But I don't use them for navigation myself, so I won't promise to add new ones myself.  ;)  I use Imenu to jump to sections and definitions, and and my [deffy](https://github.com/alphapapa/taxy.el/blob/master/examples/deffy.el) package to jump to definitions across the project, which to some extent allows me to not think about which file I'm in and where within one.

In case it's helpful to anyone:
```el
(use-package imenu
  :config
  (defun ap/emacs-lisp-imenu-hook ()
    "Add entry to `imenu-generic-expression' for Emacs Lisp buffers."
    ;; TODO: Upstream this.
    (cl-pushnew '("Sections"
                  ;; (rx bol (0+ blank) (>= 3 ";") (0+ blank) (group (1+ nonl)))
                  "^[[:blank:]]*;\\{3,\\}[[:blank:]]*\\(.+\\)"
                  1)
                imenu-generic-expression))
  (add-hook 'emacs-lisp-mode-hook #'ap/emacs-lisp-imenu-hook))
```
